src/frontend/js/components/CourseRunEnrollment/index.tsx

```diff
@@ -161,7 +166,13 @@ const CourseRunEnrollment: React.FC<CourseRunEnrollmentProps & CommonDataProps>
   const enroll = useCallback(async () => {
     dispatch({ type: ActionType.ENROLL });
     if (courseRun && currentUser) {
-      const isEnrolled = await setEnrollment().catch(() => undefined);
+      const isEnrolled = await setEnrollment().catch((error) => {
+        dispatch({
```

Comment:
I think you should use a `try...catch` here.
Furthermore there is something wrong as the action `UPDATE_CONTEXT` is always called. You should dispatch this action only when enroll request succeeded.